core/src/main/java/com/graphhopper/routing/util/DataFlagEncoder.java

```diff
@@ -156,6 +164,18 @@ public int defineWayBits(int index, int shift) {
         carBwdMaxspeedEncoder = new EncodedDoubleValue("car bwd maxspeed", shift, speedBits, speedFactor, 0, maxPossibleSpeed, true);
         shift += carBwdMaxspeedEncoder.getBits();
 
+        /* Value range: [3.0m, 5.4m] */
```

Comment:
Yes, the current implementation is quite simple and is based on the value set for the particular area that I have. The value set was obtained from a source independent of OSM that I managed to incorporate into OSM using an internal data ingestion pipeline. As such, the value format is very good (as it is under my control :).  I hope we can have more sophisticated implementation which is applicable to world wide data in later version.

For this particular PR, what I can do is investigating the value range that I can obtain from taginfo (Thanks for the reference - a great website). And see if we need more bits for the encoder and if I can enhance the parsing logic.